src/bioregistry/schema/struct.py

```diff
@@ -949,7 +949,7 @@ def is_canonical_identifier(self, identifier: str) -> Optional[bool]:
         pattern = self.get_pattern_re()
         if pattern is None:
             return None
-        return bool(pattern.match(identifier))
+        return pattern.fullmatch(identifier) is not None
```

Comment:
Pretty interesting case of how match can easily go wrong when there are conditionals in the regex, since the `$` doesn't get included in every option.

fullmatch is a nice solution!